### DISCUSSION TITLE
[QUESTION]: How to change the way multiple molecules are embedded in Chemprop

### DISCUSSION THREAD

**Discussion by Rainsmumu** [chemprop/chemprop#633] [Q&A]
**What are you trying to do?**
In Chemprop, there are two methods for embedding multiple molecules. One method is to train separate D-MPNN models for each molecule, while the other method is to train a single D-MPNN model for all molecules. Can I achieve the following functionality by modifying the code：
1. If the property I want to predict is determined by four molecules, I would like to use SMILES 1 and SMILES 2 to train one D-MPNN (referred to as D-MPNN 1), and then use SMILES 3 and SMILES 4 to train two different D-MPNNs (referred to as D-MPNN 2 and D-MPNN 3) respectively.
2. Can operations other than CAT, such as SUM, be performed on embeddings obtained from different D-MPNNs? For example, I want to perform a SUM operation on the embeddings obtained from D-MPNN 1, D-MPNN 2 and D-MPNN 3.

**Screenshots**
<img width="685" alt="Snipaste_2024-01-02_16-31-38" src="https://github.com/chemprop/chemprop/assets/102937720/23f29a5b-a998-4847-be8d-ebb80df7de69">

**Comment by cjmcgill** [comment]
Hi, neither of these are things that Chemprop can do with current configuration.

It is something that could be modified locally or have a custom branch to handle the changes as I think they'd only affect a handful of locations. I can make a branch that has some initial changes in this direction for you if you would be comfortable with making any further changes that might be needed.

**Comment by Rainsmumu** [comment]
Hi, thank you for your quick answer. It will be great if you can do that.

**Comment by kevingreenman** [comment]
I've converted this issue to a discussion, which is the forum we're planning to use going forward for these types of questions that are more general requests for advice/help or that require more extended discussion.

**Comment by cjmcgill** [comment]
Hi @Rainsmumu I'm sorry that it took me so long to circle back around to this, but I think that you have done it correctly. The code blocks that you described in #634 are exactly what I was going to do. I do not believe that there are other places where the core functionality of chemprop would be broken. As far as I can see, this would train and predict like normal.

This will not be compatible with the fingerprint function as currently written or likely the frozen layers transfer learning functions. But they could be added with further development.
